src/macaron/dependency_analyzer/java_repo_finder.py

```diff
@@ -171,24 +199,52 @@ def find_repo(gav: str, tags: list[str]) -> Iterator[str]:
         "repofinder.java", "artifact_repositories", fallback=["https://repo.maven.apache.org/maven2"]
     )
 
-    # Parse the GAV and create the URLs for its POM
-    group, artifact, version = parse_gav(gav)
-    if not group:
-        return iter([])
-    request_urls = create_urls(group, artifact, version, repositories)
-    if not request_urls:
-        return iter([])
-
-    # Try each POM URL in order, terminating early if a match is found
-    with requests.Session() as session:
-        pom = ""
-        for request_url in request_urls:
-            pom = retrieve_pom(session, request_url)
-            if pom != "":
-                break
-
-    if pom == "":
-        return iter([])
-
-    # Parse XML data and return URL
-    return parse_pom(pom, tags)
+    # Perform the following in a loop:
+    # - Create URLs for the current artifact POM
+    # - Parse the POM
+    # - Try to extract SCM metadata and return URLs
+    # - Try to extract parent information and change current artifact to it
+    # - Repeat
+    limit = 10
+    while group and artifact and version and limit > 0:
+        # Create the URLs for retrieving the artifact's POM
+        group = group.replace(".", "/")
+        artifact = artifact.replace(".", "/")
```

Comment:
Is it possible that the artifact name contains `.`? :thinking:

Reply:
It seems there is no exact naming scheme for artifact names. As they most likely represent the name of the related JAR file, or similar, we should assume any valid file name character can be used.
Source: [Maven](https://maven.apache.org/guides/mini/guide-naming-conventions.html)

Reply:
> It seems there is no exact naming scheme for artifact names. As they most likely represent the name of the related JAR file, or similar, we should assume any valid file name character can be used. Source: [Maven](https://maven.apache.org/guides/mini/guide-naming-conventions.html)

Would that mean the URL would become invalid if the artifact name includes `.`? e.g. `foo.bar.baz` would change to `foo/bar/baz` :thinking:

Reply:
I think you are right. I'll remove the replacement for artifact names.